server/asset/eth/eth.go

```diff
@@ -207,19 +207,18 @@ func (eth *Backend) Connect(ctx context.Context) (*sync.WaitGroup, error) {
 
 // TxData fetches the raw transaction data.
 func (eth *Backend) TxData(coinID []byte) ([]byte, error) {
-	cnr, err := DecodeCoinID(coinID)
+	txHash, err := decodeCoinID(coinID)
 	if err != nil {
 		return nil, fmt.Errorf("coin ID decoding error: %v", err)
 	}
-	c, is := cnr.(*TxCoinID)
-	if !is {
-		return nil, fmt.Errorf("wrong type of coin ID, %v", cnr)
-	}
 
-	tx, _, err := eth.node.transaction(eth.rpcCtx, c.TxID)
+	tx, _, err := eth.node.transaction(eth.rpcCtx, txHash)
 	if err != nil {
 		return nil, fmt.Errorf("error retrieving transaction: %w", err)
 	}
+	if tx == nil { // Possible?
+		return nil, fmt.Errorf("no transaction %s", txHash)
+	}
```

Comment:
In my limited experience no tx is an `ethereum.NotFound` err, but this is probably safe.